pipeline/runtime/runtestsV2.go

```diff
@@ -54,18 +51,23 @@ func executeRunTestsV2Step(ctx context.Context, f RunFunc, r *api.StartStepReque
 	setTiEnvVariables(step, tiConfig)
 	agentPaths := make(map[string]string)
 	if r.RunTestsV2.IntelligenceMode {
-		err := downloadJavaAgent(ctx, tmpFilePath, fs, log)
+		links, err := instrumentation.GetV2AgentDownloadLinks(ctx, tiConfig)
+		if err != nil {
+			return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to get AgentV2 URL from TI")
+		}
+
+		err = downloadJavaAgent(ctx, tmpFilePath, links[0].URL, fs, log)
 		if err != nil {
 			return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to download Java agent")
 		}
 
-		rubyArtifactDir, err := downloadRubyAgent(ctx, tmpFilePath, fs, log)
+		rubyArtifactDir, err := downloadRubyAgent(ctx, tmpFilePath, links[2].URL, fs, log)
 		if err != nil || rubyArtifactDir == "" {
 			return nil, nil, nil, nil, nil, string(optimizationState), fmt.Errorf("failed to download Ruby agent")
 		}
 		agentPaths["ruby"] = rubyArtifactDir
 
-		pythonArtifactDir, err := downloadPythonAgent(ctx, tmpFilePath, fs, log)
+		pythonArtifactDir, err := downloadPythonAgent(ctx, tmpFilePath, links[1].URL, fs, log)
```

Comment:
Link[0], Link[1], Link[2] stored in somewhere in such a way to define link[0] corresponds to java, link[1] corresponds to python, links[2] corresponds to ruby ??.

Reply:
Also what if by any chance "link" is empty, we can have length check here.

Reply:
yes, its stored correspondingly.
done